test/internal/kubernetes/utils_create.go

```diff
@@ -209,157 +223,248 @@ func InstallCNSDaemonset(ctx context.Context, clientset *kubernetes.Clientset, l
 	cniDropgzVersion := os.Getenv(envCNIDropgzVersion)
 	cnsVersion := os.Getenv(envCNSVersion)
 
-	cns, err := loadCNSDaemonset(ctx, clientset, cnsVersion, cniDropgzVersion)
+	cnsLinux, cnsLinuxDetails, err := loadCNSDaemonset(ctx, clientset, cnsVersion, cniDropgzVersion, corev1.Linux)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load linux CNS daemonset")
+	}
+
+	cnsWindows := appsv1.DaemonSet{}
+	cnsWindowsDetails := cnsDetails{}
+	hasWinNodes, err := HasWindowsNodes(ctx, clientset)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to load CNS daemonset")
+		return nil, errors.Wrap(err, "failed to check if cluster has windows nodes")
+	}
+	if hasWinNodes {
+		cnsWindows, cnsWindowsDetails, err = loadCNSDaemonset(ctx, clientset, cnsVersion, cniDropgzVersion, corev1.Windows)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to load windows CNS daemonset")
+		}
 	}
 
 	cleanupds := func() error {
-		if err := ExportLogsByLabelSelector(ctx, clientset, cns.Namespace, cnsLabelSelector, logDir); err != nil {
-			return errors.Wrapf(err, "failed to export logs by label selector %s", cnsLabelSelector)
+		if err := ExportLogsByLabelSelector(ctx, clientset, cnsLinux.Namespace, cnsLinuxDetails.labelSelector, logDir); err != nil {
+			return errors.Wrapf(err, "failed to export linux logs by label selector %s", cnsLinuxLabelSelector)
+		}
+		if hasWinNodes {
+			if err := ExportLogsByLabelSelector(ctx, clientset, cnsWindows.Namespace, cnsWindowsDetails.labelSelector, logDir); err != nil {
+				return errors.Wrapf(err, "failed to export windows logs by label selector %s", cnsWindowsLabelSelector)
+			}
 		}
```

Comment:
If I remember correctly, this will get used by a cluster that has both linux and windows CNS installed. Should we try to wrap it in both windows and linux logs or do and if/else statement? As it stands now there is a possibility that the `hasWinNodes` check will be skipped as it will return before it reaches it.

Reply:
> If I remember correctly, this will get used by a cluster that has both linux and windows CNS installed.
Correct

That's a good point, if something fails to get the linux logs, then we'd skip the windows logs all together

Reply:
Fixed